### DISCUSSION TITLE
A displaytext batch file and VARs

### DISCUSSION THREAD

**Discussion by SVLD2** [arendst/Tasmota#18348] [Q&A]
Hi all,
Due to the fact that the size of the Rules is quite limited, for a large number of **DisplayText** commands decided to try batch files.
But I need to display the values of the VARx. Is there any way to access variables from displaytext batch files?

And one more question - is there an abbreviation for the **DisplayText** command (for example, **DP**)?

**Comment by gemu2015** [comment]
check the display docs. there is an example of a display.bat file with variables.
otherwise you may switch to scripting there is a shortcut dt for displaytext.

**Reply by sfromis** [reply]
When using rules, the length of "DisplayText" does not really limit what you can have in each rule set, as such repeated words will be compressed very efficiently. Repeating that word a lot of times gave;
`Compressed from 519 to 37 (-93%)`

**Reply by SVLD2** [reply]
@gemu2015 
if you mean this documentation [Batch files](https://tasmota.github.io/docs/Displays/#batch-files), then not a word about variables is said here.
%VARx% in batch files do not work

@sfromis 
I had to heavily optimize the text to fit in the memory limit. had to fight for every letter! acronym would help me a lot(

**Reply by gemu2015** [reply]
I mean the display text CMD dv (Display variables) 

but this command only supports JSON variables from Sensors

**Reply by sfromis** [reply]
No, shortening from 11 to 2 would **not** "help you a lot" in a rule set, as it is not every time, but only once a rule set. Compression means that using "DisplayText" again costs no more space than it would if using "DP" again. You can usually get more than 1½ KB of rule text compressed into the size of ½ KB.
